### ISSUE TITLE
suggestion: xgb.cv returns pd.DataFrame or list of lists of numbers

### ISSUE THREAD

**Issue by fyears** [dmlc/xgboost#294]
To my surprise, in Python, `xgb.cv()` returns list of strings! Like:

```python
[0]     cv-test-mlogloss:1.544280+0.012040      cv-train-mlogloss:1.515598+0.010420
[1]     cv-test-mlogloss:1.274120+0.008550      cv-train-mlogloss:1.224700+0.006411
[2]     cv-test-mlogloss:1.109361+0.008391      cv-train-mlogloss:1.042920+0.005977
```

Why? I think the original numbers are better for analysis, for example, plotting the relationship between iterations and the loss.

I suggest this function returns a pd.DataFrame like (I have to write a function to manually parse the above!):

```python
   iter_from_zero  test_mlogloss  test_mlogloss_sd  train_mlogloss  \
0               0       1.544280          0.012040        1.515598   
1               1       1.274120          0.008550        1.224700   
2               2       1.109361          0.008391        1.042920   

   train_mlogloss_sd  
0           0.010420  
1           0.006411  
2           0.005977  
```

Or at least, list of lists of numbers, or similarly a  numpy.ndarray:

```python
array([[  0.00000000e+00,   1.54428000e+00,   1.20400000e-02,
          1.51559800e+00,   1.04200000e-02],
       [  1.00000000e+00,   1.27412000e+00,   8.55000000e-03,
          1.22470000e+00,   6.41100000e-03],
       [  2.00000000e+00,   1.10936100e+00,   8.39100000e-03,
          1.04292000e+00,   5.97700000e-03]])
```

**Comment by tqchen** [comment]
This is a nice suggestion. Can you open a PR for this?

**Comment by tqchen** [comment]
I think it is reasonable to return a list of numbers which can be quickly parsed by dataframe, so that we do not explicitly add the dependency on pandas. @fyears do you want to contribute on this?

Thanks!

**Comment by fyears** [comment]
@tqchen Yes. That is why I suggest this alternative output.

I think it is an easy one. I would like to have a look at the code and try in these two days.
